### PR TITLE
#598: Introduced RPC and ChainIndexing Interfaces

### PR DESCRIPTION
**Pull request by cdc-Hitesh**
- [x] interface
- [x] service business logic
- [x] EvmClient
- [x] CronosClient
- [ ] CronosClient - ChainIndexing API business logic

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,8 +7,6 @@ on:
 jobs:
   release:
     runs-on: ${{ matrix.os }}
-    env:
-      NODE_OPTIONS: '--max_old_space_size=4096'
```
Comment (crypto-3dd) [inline .github/workflows/build.yml:11]: I wonder if there is a reason you are modifying the build actions' latest updates? I would suggest to possibly bring back to the normal state from `cronos-general-support`

**Review by crypto-3dd — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -41,7 +41,6 @@
     "electron-log": "4.3.5",
     "electron-reload": "1.5.0",
     "electron-updater": "4.3.9",
-    "ethers": "^5.4.2",
```
Comment (crypto-3dd) [inline package.json:44]: Did you remove `ethers` dependency?

File: src/config/StaticAssets.ts [hunk 3]
```diff
@@ -1,40 +0,0 @@
...
-      'https://firebasestorage.googleapis.com/v0/b/chain-desktop-wallet.appspot.com/o/cronos_logo.png?alt=media&token=781c48a3-e89e-4dd4-87d3-d1a1b8e2e456',
-    identifier: getRandomId(),
-    name: 'Cronos Coin',
-    symbol: 'CRONOS',
-    mainnetSymbol: 'CRONOS', // This is to be used solely for markets data since testnet market prices is always non existent
-    stakedBalance: '0',
-    decimals: 18,
-    assetType: UserAssetType.EVM,
-  };
-};
```
Comment (crypto-3dd) [inline src/config/StaticAssets.ts:40]: I think there have been some issues when you were merging from `cronos-general-support`

File: src/config/StaticConfig.ts [hunk 4]
```diff
@@ -193,6 +193,24 @@ export const DefaultWalletConfigs = {
...
+      'https://s3-ap-southeast-1.amazonaws.com/monaco-cointrack-production/uploads/coin/colorful_logo/5c1248c15568a4017c20aa87/cro.png',
+    identifier: getRandomId(),
+    name: 'Crypto.org Coin',
+    symbol: assetSymbol,
+    mainnetSymbol: 'CRO', // This is to be used solely for markets data since testnet market prices is always non existent
+    stakedBalance: '0',
+    decimals: 8,
+  };
+};
+
```
Comment (crypto-3dd) [inline src/config/StaticConfig.ts:213]: Normally  this shouldn't be added here again

File: package.json (hunk 2 above)
Comment (cdc-Hitesh) [reply package.json:44]: These changes are somehow picked up after reverting my eslint based commits.

File: .github/workflows/build.yml (hunk 1 above)
Comment (cdc-Hitesh) [reply .github/workflows/build.yml:11]: These changes are somehow picked up after reverting my eslint based commits.

File: src/config/StaticAssets.ts (hunk 3 above)
Comment (cdc-Hitesh) [reply src/config/StaticAssets.ts:40]: I agree. Something is wrong with the merging and reverting.

File: src/config/StaticConfig.ts (hunk 4 above)
Comment (cdc-Hitesh) [reply src/config/StaticConfig.ts:213]: I agree something is wrong while reverting and merging.

**Pull request closed without merging**